### PR TITLE
Split Data East ACE Chip Emulation

### PR DESCRIPTION
**Pull request by cam900**
ACE chip(Data East 99) used on:
boogwing.cpp (Boogie Wings / The Great Ragtime Show(Japan))
deco32.cpp (Tattoo Assassins / Night Slashers)
ACE Chip is alpha blend/palette effect functions, but many unknown things.
Registers:
Byte 0x00-0x1f: alpha control value (0x0:opaque, 0x10:half transparent, 0x20:fully transparent, 0x21~:unknown)
Byte 0x20: fadeptred
Byte 0x21: fadeptgreen
Byte 0x22: fadeptblue
Byte 0x23: fadestred
Byte 0x24: fadestgreen
Byte 0x25: fadestblue
Byte 0x26: fadetype (0x1100:multiplicative, 0x1000:additive, 0x0000:normal, other:unknown)
Byte 0x27: unknown

The 'ST' value lerps between the 'PT' value and the palette entries.  So, if PT==0,
	    then ST ranging from 0 to 255 will cause a fade to black (when ST==255 the palette
	    becomes zero).
	    'fadetype' - 1100 for multiplicative fade, 1000 for additive.

Logs:
boogwing: Get rid of decocomn, Hooked up ACE Chip, improvement alpha blendings, implement color fade effects (at transporter stage, end of each level, "DANGER" effects, and some flashing effects)
deco32(Tattoo Assassins / Night Slashers): split ACE Chip emulation to mame/machine source, implement palette additive fading

### CODE REVIEW

**Comment by ajrhacker** [comment]
You forgot to `git add` the new device files, or indeed any files other than arcade.lua.

File: src/mame/drivers/boogwing.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // license:BSD-3-Clause
-// copyright-holders:Bryan McPhail, David Haywood
+// copyright-holders:Bryan McPhail, David Haywood, cam900
```
Comment (startaq) [inline src/mame/drivers/boogwing.cpp]: I don't think you should add yourself to the copyright holders list of the drivers where you only did minor changes.

**Comment by cuavas** [comment]
I'm not qualified to approve/reject the change (not familiar with this hardware), but I have a few comments:
* Please use uppercase `TODO` rather than `todo` in comments so editors like vim will highlight it.
* Is this just rearranging stuff, or does it change emulation in some way?  If the latter, please explain what it changes/fixes.
* Are there substantial changes to the code, or is it just code moved from the drivers to a device?  If it's just existing code extracted to a new device, it doesn't count as creative work that qualifies for adding to copyright holders.
* If someone does merge this, please sqaush it to avoid creating commit spam with partial changes.

**Comment by sjy96525** [comment]
I Play the Boogie Wings, Night Slashers has Graphics Problem.
https://i.imgur.com/ZLvWsi9.png
https://i.imgur.com/eEJMTlC.png

**Pull request merged**